lib/ueberauth/strategy/twitter/internal.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Ueberauth.Strategy.Twitter.OAuth.Internal do
+  @moduledoc """
+  A layer to handle OAuth signing, etc.
+  """
+
+  def get(url, extraparams, {consumer_key, consumer_secret, _}, token \\ "", token_secret \\ "") do
+    creds = OAuther.credentials(
+      consumer_key: consumer_key,
+      consumer_secret: consumer_secret,
+      token: token,
+      token_secret: token_secret
+    )
+    {header, params} = "get" |> OAuther.sign(url, extraparams, creds) |> OAuther.header
+    HTTPoison.get(url, [header], params: params)
+  end
+
+  def params_decode(resp) do
+    resp
+    |> String.split("&")
+    |> Enum.map(fn (item) -> String.split(item, "=") end)
+    |> Enum.map(&List.to_tuple/1)
```

Comment:
If we use `List.to_tuple/1` we can avoid the reduce below and instead rely on `Enum.into/2`:

```elixir
|> String.split("&", trim: true) |> Enum.map(&String.split(&1, "=")) |> Enum.map(&List.to_tuple/1) |> Enum.into(%{})
```

If we don't want to iterate through the list and create tuples, we can update the `Enum.reduce/3` to work with lists instead:

```elixir

resp
|> String.split("&", trim: true) 
|> Enum.map(&String.split(&1, "=")) 
|> Enum.reduce(%{}, fn([name, val], acc) -> Map.put_new(acc, name, val) end)
```

```elixir
iex> resp = "test=1&one=2&three=5"
"test=1&one=2&three=5"
iex> resp |> String.split("&", trim: true) |> Enum.map(&String.split(&1, "=")) |> Enum.map(&List.to_tuple/1) |> Enum.into(%{})
%{"one" => "2", "test" => "1", "three" => "5"}
iex> resp |> String.split("&", trim: true) |> Enum.map(&String.split(&1, "=")) |> Enum.reduce(%{}, fn([name, val], acc) -> Map.put_new(acc, name, val) end)
%{"one" => "2", "test" => "1", "three" => "5"}
```